src/dex-ui/hooks/governance/useAllProposals.tsx

```diff
@@ -26,11 +28,22 @@ export function useAllProposals(props: UseAllProposalsProps) {
       const doesTitleFilterMatch = proposal.title?.toLowerCase().includes(titleFilter.toLowerCase());
       const doesStatusFilterMatch = !isNil(proposal?.status) && statusFilters.includes(proposal?.status);
       if (doesTitleFilterMatch && doesStatusFilterMatch) {
-        const formattedProposal = formatProposal(proposal);
-        return [...formattedProposals, formattedProposal];
+        if (startDate && endDate) {
```

Comment:
All filter conditions need to be met for a proposal to meet the filter criteria. You can move this logic above the `if` on line 30 and add the `doesDateFilterMatch` boolean to the conditional check:

```
if (doesTitleFilterMatch && doesStatusFilterMatch && doesDateFilterMatch) {
       const formattedProposal = formatProposal(proposal);
       return [...formattedProposals, formattedProposal];
}|
```

Reply:
`"Some-Text".includes("")` this always comes as true so even if the `FilterText` is not selected `doesTitleFilterMatch ` will come true, but `dateFilter` is false by default and filter can work without date filter applied, so putting `if (doesTitleFilterMatch && doesStatusFilterMatch && doesDateFilterMatch)` in one condition will fail if dateFilter is not applied just textFilter is applied, that why I have put another condition for date filter though I have simplified the logic  now. let me know if you want to discuss in detail for this.

Reply:
Got it. The previous variables consider empty filter values a match (true). The naming is somewhat misleading.